R/run.R

```diff
@@ -186,14 +186,29 @@ run_bm <- function(bm, ..., n_iter = 1, profiling = FALSE, global_params = list(
   defaults <- lapply(get_default_args(bm$setup), head, 1)
   defaults$cpu_count <- parallel::detectCores()
   params <- modifyList(defaults, list(...))
-  params <- modifyList(params, global_params)
-  params$packages <- package_info()[, c("package", "loadedversion", "date", "source")]
-  names(params$packages)[2] <- "version"
-
-  structure(list(
-    result = do.call(rbind, results),
-    params = params
-  ), class = "arrowbench_result")
+  all_params <- modifyList(params, global_params)
+  all_params$packages <- package_info()[, c("package", "loadedversion", "date", "source")]
+  names(all_params$packages)[2] <- "version"
+  row.names(all_params$packages) <- NULL
+  # remove `packages_info` class whose print method won't work anymore
+  all_params$packages <- as.data.frame(all_params$packages)
```

Comment:
🙏 this was bugging me elsewhere recently too

Reply:
Yeah I was getting errors when trying to print and wasn't really sure if it was my fault, so I just fixed it regardless